### PR TITLE
Lecture 01-b

### PR DESCRIPTION
**Pull request by michaelgold**
The world is naturally hierarchical, thus our learning machines require multiple layers that build on each other compositionally. We define multilayer feedforward neural networks with combinations of linear and nonlinear modules using PyTorch. Then, we train these networks through stochastic gradient descent using backpropagation.

-- Second part of Lecture 1 (28 Jan 2019) by Erica Dominic and Hongyu (Florence) Lu. Edited by Michael Gold

### CODE REVIEW

**Comment by Atcold** [comment]
Tomorrow I'll check it out. Can you add the participants to the PR description?

**Comment by michaelgold** [comment]
Done. Thanks.

**Comment by Atcold** [comment]
Great. Do you think you could put the description in a format similar to https://github.com/Atcold/SP19-DL-collaborative-notes/pull/2? Thanks again!

**Comment by Atcold** [comment]
Moreover, can you add your sections as external `TEX` files, like it has been done in https://github.com/Atcold/SP19-DL-collaborative-notes/pull/2?

**Comment by Atcold** [comment]
Finally, I've fixed the formatting (see the version on Overleaf).
So, just extract one or more more `TEX` files to add to `main.tex` from your current `main.tex` and submit those, using `\chapter{}` for the main header.

**Comment by michaelgold** [comment]
> Great. Do you think you could put the description in a format similar to #2? Thanks again!

I've put our description into the format they used. I noticed that in their description they summarized the lecture into a key takeaway, whereas we gave an overview of the content. Would you prefer that we rewrite the our description in the form of a key takeaway?

**Comment by Atcold** [comment]
Yes, please. That was the aim, but I think I was not clear.
Rebase your code on top of the current `master`, and notice the current file hierarchy.
You were the first groups and I didn't plan too well. My apologies.

**Comment by michaelgold** [comment]
> Yes, please. That was the aim, but I think I was not clear.
> Rebase your code on top of the current `master`, and notice the current file hierarchy.
> You were the first groups and I didn't plan too well. My apologies.

No worries - the changes were straightforward to make :)
We rebased the code and are just making some minor formatting edits now.

**Comment by michaelgold** [comment]
Done

**Comment by Atcold** [comment]
Hmm... I don't think the rebase worked. There are 12 commits in this PR.
You need to rebase on top of the current `master`, and the `force push`.
So, you need to pull to your fork the latest updates, then do a rebase, and finally a force push.

**Comment by Atcold** [comment]
screenshot 2019-02-06 19 44 19
This is a merge. Can you please do a rebase instead?
You can find some help here: https://help.github.com/articles/about-pull-request-merges/

**Comment by michaelgold** [comment]
I think it worked this time.

**Comment by michaelgold** [comment]
Overleaf synced our work directly to the master branch of my forked repo on github. This made rebasing from master a bit trickier than usual. Now I see that I can actually use overleaf as a git remote with the command line: (https://www.overleaf.com/blog/195-new-collaborate-online-and-offline-with-overleaf-and-git-beta). I suppose it would be possible to add overleaf as a git remote and then pull the work from overleaf to a local feature branch. Then it will be easy to pull updates from your upstream repo to the local master branch, rebase from the local feature branch, and finally push to the forked repo. I'm still not sure if that's the most efficient way to do it, but if it sounds like it is, I'm happy to write it up and post on Piazza.

**Comment by Atcold** [comment]
Good job! Yes, feel free to write a post on Piazza, if you think someone will benefit from it! Perhaps copy it here as well?

**Comment by Atcold** [comment]
Holy cow! The quality of your (as a group) job is absolutely mind blowing!!!
I'll use this contribution as template for others. Thanks again so much!
